qiskit/providers/basicaer/qasm_simulator.py

```diff
@@ -387,18 +385,21 @@ def _validate_measure_sampling(self, experiment):
             # measure sampling is allowed
             self._sample_measure = True
 
-    @deprecate_arg(
-        "qobj",
-        deprecation_description="Using a qobj for the first argument to QasmSimulatorPy.run()",
-        since="0.22.0",
-        pending=True,
-        predicate=lambda qobj: not isinstance(qobj, (QuantumCircuit, list)),
-    )
-    def run(self, qobj, **backend_options):
-        """Run qobj asynchronously.
+    def run(self, run_input, **backend_options):
+        """Run on the backend.
 
         Args:
-            qobj (Qobj): payload of the experiment
+            run_input (QuantumCircuit or Schedule or list): An individual or a
+                list of :class:`~qiskit.circuit.QuantumCircuit` or
+                :class:`~qiskit.pulse.Schedule` objects to run on the backend.
+                For legacy providers migrating to the new versioned providers,
+                provider interface a :class:`~qiskit.qobj.QasmQobj` or
+                :class:`~qiskit.qobj.PulseQobj` objects should probably be
+                supported too (but deprecated) for backwards compatibility. Be
+                sure to update the docstrings of subclasses implementing this
+                method to document that. New provider implementations should not
+                do this though as :mod:`qiskit.qobj` will be deprecated and
+                removed along with the legacy providers interface.
```

Comment:
This part of the docstring doesn't look necessary here.

Reply:
Thank you once more for your assistance. I too felt the same, let me retain the original wording in the string. Your help is greatly appreciated.